### PR TITLE
macports: avoid dylib/link confusion

### PR DESCRIPTION
**Pull request by rurban**
use true bundle/dylib extensions, avoid mixup, avoid hardcoded .so,
add -L/opt/local/lib -I/opt/local/include search paths for macports,
use libglfw.dylib, not libglfw3.dylib

not necesserily to be merged, but this is what I needed

### CODE REVIEW

**Comment by eriksvedang** [comment]
Nice, making it work well with macports would be great. I get a lot of errors when running the tests after this patch though. I think it has to do with the .bundle suffix. Run the tests with ```CARP_DEV=1 carp```.

**Comment by daveyarwood** [comment]
@rurban Your instructions in `INSTALL.md` pointed me in the right direction to build Carp successfully! 🎉 

I use Homebrew, not Macports, so my libffi is in a slightly different directory. What I ran was:

```
git clone https://github.com/eriksvedang/Carp.git
cd Carp
cmake . -DLIBFFI_INCLUDE_PATH=/usr/local/opt/libffi/lib/libffi-3.0.13/include/ -DLIBFFI_LIBRARY_PATH=/usr/local/opt/libffi/lib/
make
```

File: lisp/gl.carp [hunk 1]
```diff
@@ -1,6 +1,7 @@
 
-(if (osx?) (def *LIBNAME* "libglfw3.dylib") "")
-(if (linux?) (def *LIBNAME* "libglfw.so") "")
+;(if (osx?) (def *LIBNAME* "libglfw3.dylib") "")
+;(if (linux?) (def *LIBNAME* "libglfw.so") "")
+(def *LIBNAME* (str (if (windows?) "" "lib") "glfw" (:dylib-extension platform-specifics))
```
Comment (mkfifo) [inline lisp/gl.carp]: I find this a lot harder to reason about.

Where is the '3' needed for 'libglfw3.dylib' now coming from?

I would much prefer an if for each os, as previously done.

File: bin/carp [hunk 2]
```diff
@@ -7,4 +7,4 @@ WRAPPER=rlwrap
 #WRAPPER=lldb
 #WRAPPER=valgrind
 
-CARP_DIR=$DIR/../ $WRAPPER $DIR/carp-repl $1
+CARP_DIR=$DIR/../ $WRAPPER $DIR/carp-repl $@
```
Comment (mkfifo) [inline bin/carp]: this is solid and we should merge this.

**Comment by mkfifo** [comment]
since @daveyarwood was helped by install instructions I think it is worth including those too.

**Comment by eriksvedang** [comment]
I (manually) added the changes suggested by @mkfifo and will now close this. Added you as a contributor to README.md, @rurban . Thanks for the help!

**Pull request closed without merging**